blockchain/chainio.go

```diff
@@ -1174,6 +1175,67 @@ func dbFetchUtxoEntry(dbTx database.Tx, hash *chainhash.Hash) (*UtxoEntry, error
 	return entry, nil
 }
 
+// dbFetchUxtoStats fetches statistics on the current unspent trnsaction output set.
+func dbFetchUtxoStats(dbTx database.Tx) (*UtxoStats, error) {
+	utxoBucket := dbTx.Metadata().Bucket(dbnamespace.UtxoSetBucketName)
+
+	var stats UtxoStats
+	transactions := make(map[chainhash.Hash]int64)
+	leaves := make([]chainhash.Hash, 0)
+	cursor := utxoBucket.Cursor()
+
+	for ok := cursor.First(); ok; ok = cursor.Next() {
+		key := cursor.Key()
+		hash := chainhash.HashH(key)
+		serializedUtxo := cursor.Value()
+		entrySize := len(serializedUtxo)
+
+		// A non-nil zero-length entry means there is an entry in the database
+		// for a fully spent transaction which should never be the case.
+		if entrySize == 0 {
+			return nil, AssertError(fmt.Sprintf("database contains entry "+
+				"for fully spent tx %s", hash))
+		}
+
+		stats.Utxos++
+		stats.Size += int64(entrySize)
+		_, ok := transactions[hash]
```

Comment:
This can be simplified to just:
```Go
transactions[hash]++
```

This is the case because looking up an entry that doesn't exist returns the zero value for the type which is 0 for an `int64`.